app/database/queries/breeze_buddy/user_management.py

```
@@ -0,0 +1,436 @@
+# ============================================================================
+# Validation Functions
+# ============================================================================
+
+async def check_username_exists(username: str) -> bool:
+    """Check if a username already exists."""
+    query = "SELECT 1 FROM users WHERE username = $1"
+    try:
+        async for conn in get_db_connection():
+            row = await conn.fetchrow(query, username)
+            return row is not None
+    except Exception as e:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)